### PR TITLE
Add isValid* static methods for validating Rid components

### PR DESCRIPTION
**Pull request by ellisjoe**

### CODE REVIEW

**Comment by ellisjoe** [comment]
My main question is how many test cases we really want here

File: src/main/java/com/palantir/ri/ResourceIdentifier.java [hunk 1]
```diff
@@ -247,26 +291,27 @@ public static ResourceIdentifier of(String service, String instance, String type
     }
 
     private static void checkServiceIsValid(String service) {
```
Comment (markelliot) [inline src/main/java/com/palantir/ri/ResourceIdentifier.java:293]: I wonder if we should make these public

**Comment by markelliot** [comment]
think this is fine

File: src/main/java/com/palantir/ri/ResourceIdentifier.java (hunk 1 above)
Comment (ellisjoe) [inline src/main/java/com/palantir/ri/ResourceIdentifier.java:293]: Possibly. I know we need to extra checking and our own exception messages so we won't be using them.

Comment (markelliot) [inline src/main/java/com/palantir/ri/ResourceIdentifier.java:293]: ah right, we probably want something like `#checkServiceMatches(String)` and `#checkInstanceMatches(String)` which are instance methods and produce reasonable error messages when there's a mismatch.

**Pull request merged**